clustering/common/src/main/java/org/jboss/as/clustering/controller/RemoveStepHandler.java

```diff
@@ -72,28 +75,44 @@ protected boolean requiresRuntime(OperationContext context) {
     protected void performRemove(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
         Resource resource = context.readResource(PathAddress.EMPTY_ADDRESS);
         // Determine whether super impl will actually remove the resource
-        boolean remove = !resource.getChildTypes().stream()
-                .anyMatch(type -> resource.getChildren(type).stream()
-                        .filter(entry -> !entry.isRuntime())
-                        .map(entry -> entry.getPathElement())
-                        .anyMatch(path -> resource.hasChild(path)));
+        boolean remove = false;
```

Comment:
Unless I'm mistaken, I think you've inadvertently inverted the above logic.  "remove" should initially be true, and set to false if there are any remaining non-runtime children.